File: src/EShop.Data/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace EShop.Data.Entities
{
    public class Order
    {
        public int OrderId { get; set; }
        public DateTime Date { get; set; }
        public string Status { get; set; } = "New";
        public virtual ICollection<ProductOrder> ProductOrders { get; set; }

        public Customer Customer { get; set; }

        public int PaymentMethodId { get; set; }
        public PaymentMethod PaymentMethod { get; set; }

        public int? DeliveryMethodId { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
    }
}

File: src/EShop.App.Web/ClientApp/src/app/models/order.ts
import { Customer } from "./customer";
import { PaymentMethod } from "./paymentMethod";
import { DeliveryMethod } from "./deliveryMethod";
import { PickupPoint } from "./pickupPoint";
import { ProductOrder } from "./productOrder";

export class Order {
    orderId: number;
    date: Date;
    status: string;
    comment: string;
    customerId: number;
    customer: Customer;
    paymentMethodId: number;
    deliveryMethodId: number;
    pickupPointId: number;
    // paymentMethod: PaymentMethod;
    // deliveryMethod: DeliveryMethod;
    // pickupPoint: PickupPoint;
}